src/main/kotlin/cacophony/codegen/BlockLabel.kt

```diff
@@ -1,3 +1,23 @@
 package cacophony.codegen
 
+import cacophony.semantic.syntaxtree.Definition
+
 data class BlockLabel(val name: String)
+
+object FunctionBodyLabel {
```

Comment:
I do not think we should use singletons to implicitly pass the id of the function. 

Another idea: function handler is already quite a bloated class, so why not make it also responsible for generating asm labels. FunctionHandler.generateFunctionHandlers looks like a good place for that. We have the list of all functions to take the ids, and list of the ancestors, so we can easily make the label depend on the labels of the parents (if we choose to do so in the future, which would be ideal imo, but it is up for discussion when we introduce the FFI).

This would also have the benefit that the Call CFGNode could contain just a string label. As of now it contains a direct link to AST node, which is quite strange in my opinion.

Reply:
I do not want to approve mutable singletons storing global state. Maybe just leave go back to the solution with hashes for now.

Reply:
I think some lambdas in C++ are mangled using a hash, so I will go back to that approach.

We might want to implement a proper name mangling if we decide to go with functional direction in the future.